### ISSUE TITLE
Error on build with default config

### ISSUE THREAD

**Issue by miket2marcom** [netlify/plugin-csp-nonce#51]
I tested this plugin out on one of our Nuxt static sites, and the build triggered by enabling the plugin failed with the following error:

`netlify/edge-functions/__csp-nonce.ts(4,1): error TS2578: Unused "@ts-expect-error" directive.`

This was from a test with the default plugin configuration options - I didn't change anything. Turning off the plugin lets the site build fine again.

I couldn't find much info at all regarding this error, and the __csp-nonce.ts file being called out is not in our project.

For what it's worth, I also have an edge function on this site which sets the other security headers. In /netlify/edge-functions, I have this set-response-headers.ts:

```
/*
https://github.com/netlify/edge-functions-examples/blob/main/netlify/edge-functions/set-response-header.ts
*/

import type { Context } from 'https://edge.netlify.com'

export default async (request: Request, context: Context) => {
  const url = new URL(request.url)

  context.log(`Adding a custom header to the response for ${url}`)

  const response = await context.next()

  const resClone = response.clone()
  const resBod = await resClone.text()

  if (resBod.includes('<!DOCTYPE html>')) {
    const newBod = resBod
    context.log(newBod)
    const blob = new Blob([newBod], { type: 'text/html' })
    const newRes = new Response(blob)
    newRes.headers.set('X-Frame-Options', 'SAMEORIGIN') // add X-Frame-Options
    newRes.headers.set('X-Content-Type-Options', 'nosniff') // add X-Content-Type-Options
    newRes.headers.set('Referrer-Policy', 'strict-origin-when-cross-origin') // add Referrer-Policy
    newRes.headers.set(
      'Permissions-Policy',
      'accelerometer=*, ambient-light-sensor=*, autoplay=*, battery=*, camera=*, cross-origin-isolated=*, display-capture=*, document-domain=*, encrypted-media=*, execution-while-not-rendered=*, execution-while-out-of-viewport=*, fullscreen=*, geolocation=*, gyroscope=*, keyboard-map=*, magnetometer=*, microphone=*, midi=*, navigation-override=*, payment=*, picture-in-picture=*, publickey-credentials-get=*, screen-wake-lock=*, sync-xhr=*, usb=*, web-share=*, xr-spatial-tracking=*',
    )
    return newRes
  } else {
    response.headers.set('X-Frame-Options', 'SAMEORIGIN') // add X-Frame-Options
    response.headers.set('X-Content-Type-Options', 'nosniff') // add X-Content-Type-Options
    response.headers.set('Referrer-Policy', 'strict-origin-when-cross-origin') // add Referrer-Policy
    response.headers.set(
      'Permissions-Policy',
      'accelerometer=*, ambient-light-sensor=*, autoplay=*, battery=*, camera=*, cross-origin-isolated=*, display-capture=*, document-domain=*, encrypted-media=*, execution-while-not-rendered=*, execution-while-out-of-viewport=*, fullscreen=*, geolocation=*, gyroscope=*, keyboard-map=*, magnetometer=*, microphone=*, midi=*, navigation-override=*, payment=*, picture-in-picture=*, publickey-credentials-get=*, screen-wake-lock=*, sync-xhr=*, usb=*, web-share=*, xr-spatial-tracking=*',
    )
    return response
  }
}
```

The above works fine on its own and properly sets all the other security headers, but I'm wondering if the CSP plugin may not be playing nice with this.

Any help here would be greatly appreciated. I'd love to get this working.

Thanks!

**Comment by FiveOFive** [comment]
Trying this out and had the same issue as well. No other edge functions in our case.

**Comment by jasonbarry** [comment]
@miket2marcom @FiveOFive this should be fixed as of version 1.2.3, can you please confirm?

**Comment by FiveOFive** [comment]
> @miket2marcom @FiveOFive this should be fixed as of version 1.2.3, can you please confirm?

It built for us. Thanks, Jason. Excited to try this out!

**Comment by miket2marcom** [comment]
> @miket2marcom @FiveOFive this should be fixed as of version 1.2.3, can you please confirm?

Just now realizing I meant to post this issue in the plugin's [integration-csp](https://github.com/netlify/integration-csp) repository. Is the plugin updated as well?

**Comment by jasonbarry** [comment]
No worries @miket2marcom -- I just merged https://github.com/netlify/integration-csp/pull/23, so the next deploy of any site using the integration should pull in version 1.2.3

**Comment by miket2marcom** [comment]
Thanks for jumping on this Jason! I've enabled the plugin and now my site builds fine without error. But I don't see the CSP being applied on the new build - there's no nonce in the source and sites like securityheaders.com aren't recognizing a CSP. 

The only thing I changed in the settings was to set "Report Only" to False, in order to see the Content-Security-Policy header, but I'm still not seeing it.
